### PR TITLE
New Email system

### PR DESCRIPTION
**Pull request by lonnieezell**

### CODE REVIEW

File: system/Mail/MessageInterface.php [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+     * Example:
+     *
+     *  $message = new App\Mail\UserWelcome([
+     *      'to' => 'John Doe <john.doe@example.com>',
+     *      'from' => 'Jane Doe <jane.doe@example.com>'
+     *  ]);
+     *
+     * @param array|null $options
+     */
+    public function __construct(array $options=null);
```
Comment (samsonasik) [inline system/Mail/MessageInterface.php]: I think interface should not define `__construct()`, we may need an abstract class for `__construct()` requirement.

**Comment by baselbj** [comment]
do you need any help in this?

**Comment by lonnieezell** [comment]
@baselbj thanks for the offer, but nope. Think I finally have something I like. Third time's a charm, right? :)

If you're looking for a bigger job to tackle, then #114 or #141 would be next steps after this. Or porting over SQLite or MsSQL. Just let me know which one sounds like something you'd be interested in and we can discuss details.

File: system/Common.php [hunk 2]
```diff
@@ -872,3 +872,48 @@ function slash_item($item)
...
+    {
+        static $suhosinFuncBlacklist;
+
+        if (function_exists($functionName))
+        {
+            if ( ! isset($suhosinFuncBlacklist))
+            {
+                $suhosinFuncBlacklist = extension_loaded('suhosin')
+                    ? explode(',', trim(ini_get('suhosin.executor.func.blacklist')))
+                    : array();
```
Comment (samsonasik) [inline system/Common.php]: short array syntax

**Pull request closed without merging**